### PR TITLE
Networking Generics Refactor

### PR DESCRIPTION
**Pull request by jonesandcode**
## Reason
Trello Card: [Refactor Networking using Generics](https://trello.com/c/y9XC83wN)

## What I Did
Refactored my functions to support generics. I now only have two functions, one for my network request and one to convert my data to my WeatherModel data type.

## How I Did It
Removed the old functions and created one generic network request function that can take in any type of object I'd like.

I also added a fetchWeather function for my specific weather API call that turns my generic object into a WeatherModel object.

## How You Can Test It
I have three labels hooked up in ViewController.swift with an update labels function. Feel free to change any of those labels to a different property in my WeatherModel and see if you get the correct response.

## Screenshots
simulator screen shot - iphone x - 2018-10-11 at 10 37 50

### CODE REVIEW

File: weather/weather/ViewController.swift [hunk 1]
```diff
@@ -13,13 +13,32 @@ class ViewController: UIViewController {
...
     
     // MARK: Lifecycle Methods
     
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
-        weatherNetworkRequest.loadWeatherData()
+        
+       weatherNetworkRequest.fetchWeather(url: "https://api.darksky.net/forecast/\(weatherKey.apiKey)/42.2808256,-83.7430378") { (weather, error) in
+        if let weather = weather {
```
Comment (jshier) [inline weather/weather/ViewController.swift]: Indentation's a little funky here. Also, it looks like you'll always call `print(error)`, as it's not inside an `else` case.

File: weather/weather/ViewController.swift [hunk 2]
```diff
@@ -13,13 +13,32 @@ class ViewController: UIViewController {
     // MARK: Properties
     
     let weatherNetworkRequest = WeatherNetworkRequest()
+    let weatherKey = WeatherKey()
+    
+    @IBOutlet weak var testOutlet: UILabel!
```
Comment (jshier) [inline weather/weather/ViewController.swift]: I would suggest naming your variables properly even when you're not going to use them in the future. It's a good habit, and leads to less confusion in the moment.

File: weather/weather/ViewController.swift (hunk 1 above)
Comment (jonesandcode) [reply weather/weather/ViewController.swift]: good catch, changes have been made.

File: weather/weather/ViewController.swift (hunk 2 above)
Comment (jonesandcode) [reply weather/weather/ViewController.swift]: Excellent suggestion, changed my outlet names to what they represent on the screen.

**Pull request merged**